internal/credentials/credentials.go

```diff
@@ -119,6 +124,39 @@ func (k *kubernetesDatabase) Get(
 		); err != nil {
 			return creds, false, err
 		}
+
+		if secret == nil {
```

Comment:
I think this whole block should be outdented one level from where it is. i.e. Close the previous `if secret == nil` block before this block.

Reply:
Mainy for readability.